python/tree_plotting.py

```diff
@@ -409,75 +452,79 @@ def draw_plot(plotDict, plotname, lumi, year, logY=True, logX=False, plotData=Fa
+  h_axis = ROOT.TH1F("h_axis","", MCplot.GetNbinsX(), MCplot.GetXaxis().GetBinLowEdge(1), MCplot.GetXaxis().GetBinUpEdge(MCplot.GetNbinsX())) if is1Dplot else \
+           ROOT.TH2F("h_axis","", MCplot.GetNbinsX(), MCplot.GetXaxis().GetBinLowEdge(1), MCplot.GetXaxis().GetBinUpEdge(MCplot.GetNbinsX()),
+                                  MCplot.GetNbinsY(), MCplot.GetYaxis().GetBinLowEdge(1), MCplot.GetYaxis().GetBinUpEdge(MCplot.GetNbinsY()))
   h_axis_ratio = ROOT.TH1F("h_axis_ratio","", MCplot.GetNbinsX(), MCplot.GetXaxis().GetBinLowEdge(1), MCplot.GetXaxis().GetBinUpEdge(MCplot.GetNbinsX()))
   if logX and MCplot.GetXaxis().GetBinLowEdge(1) < epsilon:
     h_axis.GetXaxis().SetRangeUser(MCplot.GetXaxis().GetBinCenter(1)-0.25*MCplot.GetXaxis().GetBinWidth(1), MCplot.GetXaxis().GetBinUpEdge(MCplot.GetNbinsX()))
     h_axis_ratio.GetXaxis().SetRangeUser(MCplot.GetXaxis().GetBinCenter(1)-0.25*MCplot.GetXaxis().GetBinWidth(1), MCplot.GetXaxis().GetBinUpEdge(MCplot.GetNbinsX()))
 
   doRatio=False
-  if plotData:
-    if not args.dataOnly:
+  if is1Dplot:
+    if plotData:
+      if not args.dataOnly:
+        doRatio=True
+
+      #plotUtils.ConvertToPoissonGraph(curPlots["Data"], g_data, drawZeros=True, drawXerr=False)
+      plotUtils.ConvertToPoissonGraph(curPlots["Data"], g_data, drawZeros=False, drawXerr=False)
+      g_data.SetMarkerStyle(20)
+      g_data.SetMarkerSize(1.2)
+      g_data.SetLineWidth(1)
+      # draw with zero marker size so error bars drawn all the way to x axis in the case of 0 content
+      g_data_clone = g_data.Clone()
+      g_data_clone.SetMarkerSize(0.0)
+
+      #plotUtils.GetPoissonRatioGraph(MCplot, curPlots["Data"], g_ratio, drawZeros=True, drawXerr=False, useMCErr=False)
+      plotUtils.GetPoissonRatioGraph(MCplot, curPlots["Data"], g_ratio, drawZeros=False, drawXerr=False, useMCErr=False)
+      g_ratio.SetMarkerStyle(20)
+      g_ratio.SetMarkerSize(1.2)
+      g_ratio.SetLineWidth(1)
+
+    if not plotData and args.shape and doSignalMCRatio:
       doRatio=True
 
-    #plotUtils.ConvertToPoissonGraph(curPlots["Data"], g_data, drawZeros=True, drawXerr=False)
-    plotUtils.ConvertToPoissonGraph(curPlots["Data"], g_data, drawZeros=False, drawXerr=False)
-    g_data.SetMarkerStyle(20)
-    g_data.SetMarkerSize(1.2)
-    g_data.SetLineWidth(1)
-    # draw with zero marker size so error bars drawn all the way to x axis in the case of 0 content
-    g_data_clone = g_data.Clone()
-    g_data_clone.SetMarkerSize(0.0)
-
-    #plotUtils.GetPoissonRatioGraph(MCplot, curPlots["Data"], g_ratio, drawZeros=True, drawXerr=False, useMCErr=False)
-    plotUtils.GetPoissonRatioGraph(MCplot, curPlots["Data"], g_ratio, drawZeros=False, drawXerr=False, useMCErr=False)
-    g_ratio.SetMarkerStyle(20)
-    g_ratio.SetMarkerSize(1.2)
-    g_ratio.SetLineWidth(1)
-
-  if not plotData and args.shape and doSignalMCRatio:
-    doRatio=True
-
-    for i,sample in enumerate(plotDict.keys()):
```

Comment:
I'm a little confused here about the git diff.. What is the difference between L443-L480, and the new change L490-L527?

Reply:
Indentation, because there is another `if` clause wrapping these lines in the new version of the code.